### PR TITLE
Doc/renderer extension guide

### LINKED ISSUE #1240: Docs: Extension Development - Extension Guides

### PR DESCRIPTION
**Pull request by jim-docker**
so far covers
- cluster pages
- cluster page menus
- global pages
- global page menus

rest is marked WIP

addresses #1240

### CODE REVIEW

File: docs/extensions/guides/renderer-extension.md [hunk 1]
```diff
@@ -1 +1,425 @@
...
+    return (
+      <div>
+        <p>Help</p>
+      </div>
+    )
+  }
+}
+```
+
+`HelpIcon` introduces one of Lens' built-in components available to extension developers, the `Component.Icon`. Built in are the [Material Design](https://material.io) icons. One can be selected by name via the `material` field.
```
Comment (chenhunghan) [inline docs/extensions/guides/renderer-extension.md]: Can link directly to the icon page? <https://material.io/resources/icons/>

File: docs/extensions/guides/renderer-extension.md [hunk 2]
```diff
@@ -1 +1,425 @@
...
+    console.log('custom renderer process extension code started');
+  }
+
+  onDeactivate() {
+    console.log('custom renderer process extension de-activated');
+  }
+}
+```
+
+There are two methods that you can implement to facilitate running your custom code. `onActivate()` is called when your extension has been successfully enabled. By implementing `onActivate()` you can initiate your custom code. `onDeactivate()` is called when the extension is disabled (typically from the [Lens Extensions Page]()) and when implemented gives you a chance to clean up after your extension, if necessary. The example above simply logs messages when the extension is enabled and disabled.
```
Comment (chenhunghan) [inline docs/extensions/guides/renderer-extension.md:23]: Url is missing? `[Lens Extensions Page]())` or the page is not exist yet?

**Review by chenhunghan — APPROVED** [review]
LGTM

File: docs/extensions/guides/renderer-extension.md (hunk 2 above)
Comment (jim-docker) [reply docs/extensions/guides/renderer-extension.md:23]: I assumed it doesn't exist yet (if ever?)

File: docs/extensions/guides/renderer-extension.md (hunk 1 above)
Comment (jim-docker) [reply docs/extensions/guides/renderer-extension.md]: could, but the page has a link to them...

**Pull request merged**